src/System.CommandLine/ParseResult.cs

```diff
@@ -162,17 +178,41 @@ public IReadOnlyList<string> UnmatchedTokens
         /// <exception cref="ArgumentException">Thrown when there was no symbol defined for given name for the parsed command.</exception>
         /// <exception cref="InvalidCastException">Thrown when parsed result can not be cast to <typeparamref name="T"/>.</exception>
         public T? GetValue<T>(string name)
-            => RootCommandResult.GetValue<T>(name);
+        {
+            var symbol = GetSymbolByName(name, valuesOnly: true);
+            return symbol switch
+            {
+                CliArgument<T> argument => GetValue(argument),
+                CliOption<T> option => GetValue(option),
+                _ => throw new InvalidOperationException("Unexpected symbol type")
+            };
+        }
 
         // TODO: diagramming
         /*
         /// <inheritdoc />
         public override string ToString() => ParseDiagramAction.Diagram(this).ToString();
         */
 
-        public ValueResult? GetValueResult(CliSymbol symbol) 
-            => valueResultDictionary.TryGetValue(symbol, out var result) 
-                ? result 
+        /// <summary>
+        /// Gets the ValueResult, if any, for the specified option.
+        /// </summary>
+        /// <param name="option">The option for which to find a result.</param>
+        /// <returns>A result for the specified option, or <see langword="null"/> if it was not entered by the user.</returns>
+        public ValueResult? GetValueResult(CliOption option)
+            => GetValueResultInternal(option);
+
+        /// <summary>
+        /// Gets the result, if any, for the specified argument.
+        /// </summary>
+        /// <param name="argument">The argument for which to find a result.</param>
+        /// <returns>A result for the specified argument, or <see langword="default"/> if it was not entered by the user.</returns>
+        public ValueResult? GetValueResult(CliArgument argument)
+            => GetValueResultInternal(argument);
```

Comment:
Is it intentional that in the comments on these, one uses `langword="default"` and one `langword="null"` when it seems that both call the same method that currently returns null if no result?